autoload/copilot_chat/buffer.vim

```diff
@@ -97,3 +169,46 @@ function! copilot_chat#buffer#welcome_message() abort
   call appendbufline(g:active_chat_buffer, 0, 'Welcome to Copilot Chat! Type your message below:')
   call copilot_chat#buffer#add_input_separator()
 endfunction
+
+function! copilot_chat#buffer#set_active(bufnr) abort
+  let l:bufnr = a:bufnr
+  if l:bufnr ==# ''
+    let l:bufnr = bufnr('%')
+  endif
+
+  if g:active_chat_buffer == l:bufnr
+    return
+  endif
+
+  let bufinfo = getbufinfo(l:bufnr)
+  if empty(bufinfo)
+    echom 'Invlid buffer number'
+    return
+  endif
+
+  " Check if the buffer is valid
+  if getbufvar(l:bufnr, '&filetype') !=# 'copilot_chat'
+    echom 'Buffer is not a Copilot Chat buffer'
+    return
+  endif
+
+  " Set the active chat buffer to the current buffer
+  let g:active_chat_buffer = l:bufnr
+endfunction
+
+function! copilot_chat#buffer#on_delete(bufnr) abort
+  if g:copilot_chat_zombie_buffer != -1
+	let l:bufinfo = getbufinfo(g:copilot_chat_zombie_buffer)
+	if !empty(l:bufinfo) " Check if the buffer wasn't wiped out by the user
+		execute 'bwipeout' . g:copilot_chat_zombie_buffer
```

Comment:
not a huge fan of this zombie_buffer but I don't have a good enough understanding of why thins might be lingering after deleting the buffer. will investigate more post-merge

Reply:
A bit late here but if I understand correctly what this is aiming to achieve, `setlocal bufhidden=delete` (or `bufhidden=wipe`, though that may be a little aggressive) should do the trick instead of this.